src/main/java/core/basesyntax/service/transactionservice/TransactionDtoServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.transactionservice;
+
+import core.basesyntax.model.dto.TransactionDto;
+import core.basesyntax.model.operations.Operation;
+import core.basesyntax.model.product.Fruit;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransactionDtoServiceImpl implements TransactionDtoService {
+    private static final String SPLITTER = ",";
+    private static final String VALID_REGEX = "[bspr],[a-z]+,[0-9]+";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int AMOUNT_INDEX = 2;
+
+    @Override
+    public List<TransactionDto> transform(List<String> data) {
+        List<TransactionDto> transactionDataEntityList = new ArrayList<>();
+        for (String line : data) {
+            if (!checkLineIsValid(line)) {
+                continue;
```

Comment:
ideally, it's better to throw exception if input report is invalid

Reply:
yeap, I asked Bohdan, he said do this case in that way (just skip invalid string)